### ISSUE TITLE
`clear()` doesn't return to default values

### ISSUE THREAD

**Issue by lovincyrus** [immutable-js/immutable-js#1565]
### What happened

Issue with `clear()` — https://facebook.github.io/immutable-js/docs/#/Record/clear

When `clear()`, it doesn't return to its default values.

### How to reproduce

```
const User = Record({
  roles: List()
})

const assertEqual = (a, b) => {
  if (a.equals(b)) return true
  throw new Error('Failed equality check')
}

const user_0 = new User({ roles: ['superuser', 'admin'] })
const user_1 = user_0.clear();

assertEqual(user_1.roles, List())

const user_3 = user_0.withMutations(mutable => mutable.reset())

assertEqual(user_3.roles, List())
```

**Comment by jdeniau** [comment]
Thank you for your bug report. The immutable-js oss fork will soon be merged and a fix for this issue has been included in #1833. Once this PR is merged, this issue will be resolved in the main branch. We will then do our best to to release the `4.0.0` version.

Commit reference: https://github.com/immutable-js/immutable-js/pull/1833/commits/55011d0314a578392584a7de3717c431ea337ea9

**Comment by jdeniau** [comment]
Hi!
The 4.0.0-rc.14 version has been released and should fix this issue.
Feel free to test it and re-open this issue if the issue is still present.
Thank you!
